processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java

```diff
@@ -74,9 +77,10 @@ public ExtractionFn getExtractionFn()
     byte[] dimensionBytes = StringUtils.toUtf8(dimension);
     byte[] valueBytes = StringUtils.toUtf8(value);
 
-    return ByteBuffer.allocate(1 + dimensionBytes.length + valueBytes.length)
+    return ByteBuffer.allocate(2 + dimensionBytes.length + valueBytes.length)
                      .put(DimFilterCacheHelper.EXTRACTION_CACHE_ID)
                      .put(dimensionBytes)
+                     .put((byte) 0xF)
```

Comment:
0xF is a valid control character. All other cache keys use 0xFF as separator which is guaranteed to never show up in any valid UTF-8 stream.